### PR TITLE
tolerate missing image pull secrets:

### PR DESCRIPTION
**Pull request by deads2k**
This relaxes the requirement that referenced `imagePullSecrets` must exist and be retrievable in order for the node to try to sync the pod.  I don't feel very strongly about this one.  Some might say it's good to relax constraints that aren't absolutely necessary, but I'm more of the fail early and fail obviously crowd.

It could lead to unexpected pull failures because of the missing secret references, but I figured I'd make it an easy option if you want it.  

@erictune @liggitt

### CODE REVIEW

**Comment by liggitt** [comment]
since that list can get populated during admission, and we don't let you modify the podspec after creation, I'm ok tolerating some of the secrets not being retrievable

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1287,7 +1286,8 @@ func (kl *Kubelet) getPullSecretsForPod(pod *api.Pod) ([]api.Secret, error) {
 	for _, secretRef := range pod.Spec.ImagePullSecrets {
 		secret, err := kl.kubeClient.Secrets(pod.Namespace).Get(secretRef.Name)
 		if err != nil {
-			return nil, err
+			glog.Warningf("Unable to retrieve pull secret %s/%s for %s/%s due to %v.  The image pull may not succeed.", pod.Namespace, secretRef.Name, pod.Namespace, pod.Name, err)
```
Comment (liggitt) [inline pkg/kubelet/kubelet.go]: maybe just tolerate not found errors?

Comment (deads2k) [inline pkg/kubelet/kubelet.go]: > maybe just tolerate not found errors?

The real distinction being?  If it fails, syncPod will try again later, right?  Other types of errors should be transient or the whole kubelet is going to have trouble, right?

**Comment by liggitt** [comment]
lgtm

**Comment by dchen1107** [comment]
@deads2k I am with you on failing early and failing obviously, but on this particular case, I think I can tolerate relaxing constraints a little bit. But I have a question on pulling failure due to missing secrets since I never encounter such failure before: does docker pullImage call returns us the proper reason on this kind failure, so that the user know why here? 

Otherwise LGTM

**Comment by liggitt** [comment]
any secrets that were found would be part of the authentication when doing the pull... if none of them successfully authenticated the pull, the error would be a "permission denied" error on the pull, with no mention of the missing secret.

**Comment by dchen1107** [comment]
Ok,  "permission denied" should be sufficient in this case for the introspection. LGTM

**Pull request merged**